tests/handlers/test_typing.py

```diff
@@ -163,7 +164,7 @@ def test_started_typing_local(self):
 
         self.assertEquals(self.event_source.get_current_key(), 0)
 
-        self.successResultOf(
+        self.get_success(
```

Comment:
these aren't quite the same thing. `successResultOf` asserts that the thing has already completed; `get_success` does not do that. I think `tests.test_utils.get_awaitable_result` should do what you want here.

That said, it probably doesn't matter here.

Reply:
`get_success` seems to just handle if it is an `awaitable`, wrap it in a `ensureDeferred` and then call `successResultOf`:

https://github.com/matrix-org/synapse/blob/03619324fc18632a2907ace4d3e73f3c4dd0b05e/tests/unittest.py#L443-L449

So I think it does the same thing?

Reply:
The difference is the `self.pump`.